extension/memorylimiterextension/factory_test.go

```diff
@@ -37,7 +38,10 @@ func TestCreate(t *testing.T) {
 	pCfg.MemorySpikeLimitMiB = 1907
 	pCfg.CheckInterval = 100 * time.Millisecond
 
-	tp, err := factory.Create(context.Background(), extensiontest.NewNopSettings(), cfg)
+	set := extensiontest.NewNopSettings()
+	set.ID = component.NewID(factory.Type())
```

Comment:
Should we switch `extensiontest.NewNopSettings()` to accept a type? Also we can generate a helper like `metadatatest.NewNopSettings` for components that calls the extensiontest but passes the type